src/API/NoteActions.php

```diff
@@ -27,7 +27,7 @@ class NoteActions extends Notes {
 	public function register_routes() {
 		register_rest_route(
 			$this->namespace,
-			'/' . $this->rest_base . '/(?P<note_id>[\d-]+)/action/(?P<action_id>[\d-]+)',
+			'/' . $this->rest_base . '/(?P<note_id>[\d-]+)/action/(?P<action_id>[\d-]+)/screen/(?P<screen>[a-z_-]+)',
```

Comment:
If for some reason we can't use the referer, I think it might be better to include these params in the body of the request as opposed to the URL.